### PR TITLE
Fixed APCs getting to a state where they can't be deconstructed

### LINKED ISSUE #19597: Can't remove power cells from Broken APCs
(WEB REPORT BY: subs455 REMOTE: 172.93.109.202:7777)
# Revision
90f2f90d2e60e2c58c789cd5d6307c6a4ce26012

# Description
I was trying to fix an broken APC, and the battery wouldnt come out, making it impossible to fix.

Me and Soose tried to troubleshoot it but he ended up just deleting it so i could make a new APC from scratch.
# Steps to Reproduce
Remove the broken APC cover with a Crowbar. The powercell cant be removed.
# What you Expected
The powercell to come out, allowing me access to the Circuit Board.
# What Actually Happened
The battery wouldn't come out.

### LINKED ISSUE #20283: Damaged APCs can sometimes not be deconstructed(?)
#### Description of issue

image
APCs that have their covers blown off can SOMETIMES not be deconstructed.
The cell is stuck in there and there's not a single tool that works on it (though you can open the wire menu and snip the wires, but it doesn't seem to do anything)
Screwdriver gave me `Close the APC first.` and a crowbar gave me `Unexpose the wires first!`(?)
However, spawning in a mommi and using a wrench on it somehow removed the cell?
I probably caused this in #19040. I haven't been able to reproduce it locally, hence no report/fix yet.

#### Steps to reproduce
Not sure. It happens with APCs broken from either explosions or APCs. It only occurs when the cover is already off.

#### Length of time in which bug has been known to occur
Definitely since #19040, not sure if before.

#### Server revision
0732f797e3e5915fcf820e1765b84c813351a755

[bugfix]

### PR DESCRIPTION
**Pull request by BarneyGumball**
[bugfix]

Fixes #20283
Fixes #19597

If APCs had their cover torn off with the maintenance panel open, they would get stuck in a state where you could no longer deconstruct them.  I've changed them so that APCs can have either their maintenance panel open, or their cover open, but not both.  This stops the hacking/maintenance panel code interfering with construction code.

I also cleaned up a previous fix that made it so you could only open the APC menu if the maintenance panel was closed.

:cl:
 * bugfix: Fixed APCs sometimes not being deconstructable.

### CODE REVIEW

**Comment by SonixApache** [comment]
thank you for unfucking my shitcode

File: code/modules/power/apc.dm [hunk 1]
```diff
@@ -626,12 +625,9 @@
...
 
 /obj/machinery/power/apc/attack_hand(mob/user)
-//	if (!can_use(user)) This already gets called in interact() and in topic()
-//		return
-	if(!user)
+	if (!can_use(user))
 		return
-	if(wiresexposed && !issilicon(user))
-		to_chat(user, "Unexpose the wires first!")
+	if(!user)
```
Comment (DamianX) [inline code/modules/power/apc.dm]: if `user` is false `can_use` will runtime, so this is useless as it is

Comment (DamianX) [reply code/modules/power/apc.dm]: still useless because `can_use` checks it now

**Comment by DamianX** [comment]
gonfligds

**Comment by BarneyGumball** [comment]
Gimme 5s DAMN

GOD DAMN IT

**Comment by BarneyGumball** [comment]
I know why the build failed, but I'm still going to open a new branch and reapply the changes because doing these web edits without actually trying to compile it myself is really dumb.

**Pull request closed without merging**